.github/workflows/main.yml

```diff
@@ -0,0 +1,111 @@
+---
+name: CI
+
+on:
+  push:
+    branches: ["master"]
+  pull_request:
+    branches: ["master"]
+  workflow_dispatch:
+
+jobs:
+  tests:
+    name: "Python ${{ matrix.python-version }}"
+    runs-on: "ubuntu-latest"
+    env:
+      USING_COVERAGE: "3.8"
+
+    strategy:
+      matrix:
+        python-version: ["3.6", "3.7", "3.8"]
+
+    steps:
+      - uses: "actions/checkout@v2"
+      - uses: "actions/setup-python@v2"
+        with:
+          python-version: "${{ matrix.python-version }}"
+      - name: "Install dependencies"
+        run: |
+          set -xe
+          python -VV
+          python -m site
+          python -m pip install --upgrade pip setuptools wheel
+          python -m pip install --upgrade coverage[toml] virtualenv tox tox-gh-actions
+      - name: "Run tox targets for ${{ matrix.python-version }}"
+        run: |
+          set -xe
+          TOXENV=py-full python -m tox
+          TOXENV=lint python -m tox
+          TOXENV=docs python -m tox
```

Comment:
suggestion (non-blocking): it'd be ideal if we could just run `lint` and `docs` on the latest Python version rather than all of them. No biggie though.

Reply:
yeah. inside a job there is no way to "escape" the matrix, so it would have to be done in a separate job

Reply:
... or I could use `if` to just make it run on latest python

Reply:
Your call. It would be nice to separate each of the tox envs to take advantage of the parallel jobs...iirc GH allows quite a few of them (10, maybe?).